pkg/cloud/api/diff.go

```diff
@@ -41,6 +41,18 @@ func diff[T any](a, b *T, trait *FieldTraits) (*DiffResult, error) {
 	return d.result, nil
 }
 
+func diffBetweenTypes[A any, B any](a *A, b *B) (*DiffResult, error) {
```

Comment:
This isn't a diff between types, it's a diff between values.

let's call this `diffStructs`

Reply:
done